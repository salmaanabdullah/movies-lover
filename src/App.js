import React, { Suspense, lazy } from 'react';
import { ConnectedRouter } from 'connected-react-router/immutable';
import ErrorBoundary from 'src/components/ErrorBoundary';
import { Route, Switch } from 'react-router-dom';
import { history } from 'src/store/store';
import 'bootstrap/dist/css/bootstrap.min.css';
import './scss/style.scss';

const DefaultLayout = lazy(() => import('./layout'));

const App = () => (
  <ErrorBoundary>
    <ConnectedRouter history={history}>
      <Suspense fallback={'loading'}>
        <Switch>
          <Route path="/" name="Home" component={DefaultLayout} />
        </Switch>
      </Suspense>
    </ConnectedRouter>
  </ErrorBoundary>
);

export default App;
