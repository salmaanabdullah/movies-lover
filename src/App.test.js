import React from 'react';
import { shallow } from 'enzyme/build';
import App from './App';

it('mounts App without crashing', () => {
  const wrapper = shallow(<App />);
  wrapper.unmount();
});

it('expect to render app component', () => {
  expect(shallow(<App />)).toMatchSnapshot();
});
