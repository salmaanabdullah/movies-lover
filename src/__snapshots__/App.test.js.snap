// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render app component 1`] = `
<ErrorBoundary
  ErrorComponent={[Function]}
  onCatch={[Function]}
>
  <Connect(ConnectedRouterWithContext)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Suspense
      fallback="loading"
    >
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_init": [Function],
              "_payload": Object {
                "_result": [Function],
                "_status": -1,
              },
            }
          }
          name="Home"
          path="/"
        />
      </Switch>
    </Suspense>
  </Connect(ConnectedRouterWithContext)>
</ErrorBoundary>
`;
